### PR TITLE
Upgrade to upstream version 16.0.1 (closes #205)

### LINKED ISSUE #205: Upgrade to Nextcloud 18
Unfortunately, it requires PHP 7.1, which isn't the default version in current Debian... Shall we install this additional PHP version or wait for Buster porting...? :thinking:

### PR DESCRIPTION
**Pull request by JimboJoe**
## Problem
- *The application is not up-to-date with latest major upstream version*

## Solution
- *Upgrade to version 16.0.1*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [ ] **Code review** : 
- [ ] **Approval (LGTM)** : 
- [ ] **Approval (LGTM)** : 
- **CI succeeded** : 
[Build Status](https://ci-apps-hq.yunohost.org/jenkins/job/nextcloud_ynh%20PR206/)
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by JimboJoe** [comment]
On hold due the fact that Nextcloud 16 requires PHP 7.1, and default PHP version is 7.0 is current Debian version...

File: manifest.json [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
+    "version": "16.0.1~ynh1",
     "url": "https://nextcloud.com",
     "license": "AGPL-3.0",
     "maintainer": {
         "name": "YunoHost Contributors",
         "email": "apps@yunohost.org"
     },
     "requirements": {
-        "yunohost": ">= 3.5.0"
+        "yunohost": ">= 3.2.0"
```
Comment (maniackcrudelis) [inline manifest.json]: 3.5.0 is needed for some helpers

**Comment by lapineige** [comment]
> On hold due the fact that Nextcloud 16 requires PHP 7.1, and default PHP version is 7.0 is current Debian version...

Pixelfed for instance install php 7.2, aside the common version 7.0.
Would it be possible to do the same (properly) ?

**Comment by JimboJoe** [comment]
Well, everything resides in the "properly"... No "high quality" apps ever installed another version PHP before, so that would need to put some effort on this... An experimental helper already exists for this, but is it mature enough for such a used app?
So... please feel free to contribute to this topic if you feel like it! :+1:

**Comment by massimilianoLe** [comment]
> On hold due the fact that Nextcloud 16 requires PHP 7.1, and default PHP version is 7.0 is current Debian version...

I use PHP7.2 in my nextcloud nginx config from this repo `https://packages.sury.org/php/` and have no issues at all with it. I upgraded it in my installation to get some nextcloud plugins to work. Can't get PHP7.1 to work with nextcloud for some reason, but like i said no problems at all with PHP7.2+nextcloud(and yunohost).
To make it work i only had to copy the `20-nextcloud.ini ` file to php7.2 and change php7.0 to php7.2 in the nextcloud nginx conf file

**Comment by lapineige** [comment]
> Pixelfed for instance install php 7.2, aside the common version 7.0.

In case there is a proper way to do this, maybe use php 7.2, as it's a newer version and the same as Pixelfed (which as far as I know is the only other app to do this). That would avoid to potential have 3 versions installed.

(by the way, the way Pixelfed install it right now doesn't seem to be blocking any other app, including Nextcloud)

**Comment by supermamie** [comment]
I think Pilea also installs php 7.2
And at least another one installs 7.1 as it is also there on my server.

File: scripts/upgrade.d/upgrade.15.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Last available nextcloud version
+next_version="16.0.0"
```
Comment (kay0u) [inline scripts/upgrade.d/upgrade.15.sh]:
```suggestion
next_version="15.0.11"
```

File: scripts/upgrade.d/upgrade.15.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+# Nextcloud tarball checksum sha256
+nextcloud_source_sha256="4532f7028b1d9bf060f75ac4fbbde52a59ecd9c9155f3178a038d3cf3609402e"
```
Comment (kay0u) [inline scripts/upgrade.d/upgrade.15.sh]:
```suggestion
nextcloud_source_sha256="59cdde8e7a4a15606efc246e37adf6401b0b4a60f33289be8725d675b9c2ae26"
```

File: scripts/upgrade.d/upgrade.15.sh (hunk 2 above)
Comment (JimboJoe) [reply scripts/upgrade.d/upgrade.15.sh]: Hm, thanks for that, but shouldn't we make a dedicated 15.0.11 PR and keep that NextCloud 16 (and even 17!) upgrade PR open, just to indicated we've already though about it for people passing by...? :-)

Comment (kay0u) [reply scripts/upgrade.d/upgrade.15.sh]: You're right, I'll update these suggestions to keep the current nextcloud version (15.0.10)

**Comment by nathanael-h** [comment]
Hello, did some of you tested this with sury repo ? I know that this caused problems on some installation, and I wonder to know what it the risk level ...?

**Pull request merged**